build failed on DragonFlyBSD-6.4

# Bug report

### Bug description:

```
2025-02-27T04:16:40.9317730Z /usr/bin/cc -pthread  -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -I/root/.xbuilder/run/1051/auxroot/include -fPIC -fno-common -Os    -I/root/.xbuilder/run/1051/auxroot/include -fPIC -fno-common -Os     -g -std=c11 -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wstrict-prototypes -Werror=implicit-function-declaration -fvisibility=hidden  -I./Include/internal -I./Include/internal/mimalloc  -I. -I./Include -I/root/.xbuilder/run/1051/auxroot/include   -I/root/.xbuilder/run/1051/auxroot/include/tirpc -I/root/.xbuilder/run/1051/auxroot/include   -I/root/.xbuilder/run/1051/auxroot/include/tirpc  -DPy_BUILD_CORE_BUILTIN -c ./Modules/timemodule.c -o Modules/timemodule.o
2025-02-27T04:16:41.0793065Z In file included from ./Include/Python.h:64,
2025-02-27T04:16:41.0798014Z                  from ./Modules/timemodule.c:3:
2025-02-27T04:16:41.0798662Z ./Modules/timemodule.c: In function 'time_clockid_converter':
2025-02-27T04:16:41.0799495Z ./Include/pymacro.h:23:25: error: static assertion failed: "sizeof(clk_id) == sizeof(*p)"
2025-02-27T04:16:41.0800210Z  #  define static_assert _Static_assert
2025-02-27T04:16:41.0800628Z                          ^~~~~~~~~~~~~~
2025-02-27T04:16:41.0801204Z ./Include/pymacro.h:77:13: note: in expansion of macro 'static_assert'
2025-02-27T04:16:41.0801791Z              static_assert((cond), #cond); \
2025-02-27T04:16:41.0802179Z              ^~~~~~~~~~~~~
2025-02-27T04:16:41.0802658Z ./Modules/timemodule.c:201:5: note: in expansion of macro 'Py_BUILD_ASSERT'
2025-02-27T04:16:41.0803366Z      Py_BUILD_ASSERT(sizeof(clk_id) == sizeof(*p));
2025-02-27T04:16:41.0803764Z      ^~~~~~~~~~~~~~~
2025-02-27T04:16:41.1033318Z gmake: *** [Makefile:3534: Modules/timemodule.o] Error 1
2025-02-27T04:16:41.1037795Z gmake: *** Waiting for unfinished jobs....
```

fulllogs:
https://github.com/leleliu008/python-distribution/actions/runs/13558699714/job/37897846484

### CPython versions tested on:

3.13

https://www.python.org/ftp/python/3.13.2/Python-3.13.2.tgz

### Operating systems tested on:

DragonFlyBSD-6.4

<!-- gh-linked-prs -->
### Linked PRs
* gh-130634
* gh-130666
<!-- /gh-linked-prs -->

--- a/Modules/timemodule.c
+++ b/Modules/timemodule.c
@@ -187,6 +187,8 @@ time_clockid_converter(PyObject *obj, clockid_t *p)
 {
 #ifdef _AIX
     long long clk_id = PyLong_AsLongLong(obj);
+#elif defined(__DragonFly__)
+    long clk_id = PyLong_AsLong(obj);
 #else
     int clk_id = PyLong_AsInt(obj);
 #endif

